Task completed

dest1999 opened the pull request:
Асинхронно запрашиваем с заданного url указанные посты и записываем их в текстовый файл

dest1999 committed:
Асинхронно запрашиваем с заданного url посты и записываем их в текстовый файл

dest1999 committed:
Асинхронно запрашиваем с заданного url посты и записываем их в текстовый файл

dest1999 committed:
Рефакторинг: HTTPClient объявляется через using

MeatBatyr commented on HttpAsyncAwait/Program.cs:
```diff
@@ -1,12 +1,47 @@
+                tasksList.Add(GetPostByIdViaHTTPAsync(i));
+            }
+
+            await Task.WhenAll(tasksList);
```
если использовать WhenAll - можно и возвращаемое значение этого метода записать в переменную и уже по нему бежать foreach

dest1999 replied:
Да, я пробовал и так, но мне кажется в данном случае мой код более понятен

MeatBatyr commented on HttpAsyncAwait/PostDTO.cs:
```diff
@@ -0,0 +1,28 @@
+{
+    internal class PostDTO
+    {
+        public int userId { get; set; }
```
Свойства должны быть с большой буквы

dest1999 replied:
В методичке с маленькой, а C# регистрозависимый, поэтому я решил сделать имена как в методичке.
Исправлено

MeatBatyr commented on HttpAsyncAwait/Program.cs:
```diff
@@ -1,12 +1,47 @@
+            try
+            {
+                var response = await client.GetStringAsync(uri);
+                var post = JsonSerializer.Deserialize<PostDTO>(response);
```
var options = new JsonSerializerOptions()
            {
                PropertyNameCaseInsensitive = true
            };

dest1999 replied:
Исправлено

dest1999 committed:
Исправлено именование переменных

dest1999 committed:
Введено jsonOptions для десериализации постов с сайта

dest1999 committed:
Рефакторинг
Введен CancellationToken с ограничением по времени для отмены Task.
Введен перехват исключения при неполной отработке событий.
В файл записываются только успешно принятые сообщения
Метод Main сделан синхронным

dest1999 committed:
Рефакторинг
Используем Linq для формирования коллекции завершенных запросов
